src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -773,12 +775,26 @@ public void updateReferencedBigQueryDataset(
    * @param updateParams resource properties to update
    */
   public void updateControlledBigQueryDataset(
-      UUID workspaceId, UUID resourceId, UpdateResourceParams updateParams) {
+      UUID workspaceId, UUID resourceId, UpdateBqDatasetParams updateParams) {
+
+    GcpBigQueryDatasetUpdateParameters bigQueryDatasetUpdateParameters =
+        new GcpBigQueryDatasetUpdateParameters();
+
+    if (updateParams.partitionExpirationTime.isPresent())
+      bigQueryDatasetUpdateParameters.defaultPartitionLifetime(
+          updateParams.partitionExpirationTime.get());
+
+    if (updateParams.tableExpirationTime.isPresent())
+      bigQueryDatasetUpdateParameters.defaultTableLifetime(updateParams.tableExpirationTime.get());
```

Comment:
I think you can set this property to `null` to indicate it's not being updated. e.g.:
`bigQueryDatasetUpdateParameters.defaultTableLifetime(updateParams.tableExpirationTime.orElse(null));`

Reply:
Yes, simplified in latest commit.